client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +50,45 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final StreamDataRetention retentionConfig;
+
+    /**
+     * If a Stream's Retention Policy {@link RetentionPolicy} is set to 'CONSUMPTION' based,
+     * the Reader Group needs to register with the Stream as a Subscriber
+     * and periodically publish retention StreamCuts either manually or at checkpoints.
+     * This can be done by setting retentionConfig to 'CONSUMPTION_BASED_USER_STREAMCUT' or
+     * 'CONSUMPTION_BASED_AT_LAST_CHECKPOINT'.
+     * If the Stream's RetentionPolicy is not Consumption based, retentionConfig should be 'NO_IMPACT'.
+     *
+     * If the Streams' Retention Policy is set to 'CONSUMPTION', but the ReaderGroup retentionConfig is set at
+     * 'NO_IMPACT', the ReaderGroup will not get registered as a Subscriber with the Stream
+     * and retention StreamCuts cannot be published for this ReaderGroup.
+     * This will break the promise of ConsumptionBased Retention.
+     *
+     * NO_IMPACT - Read Positions of Readers do not impact Stream truncation/retention.
+     * CONSUMPTION_BASED_USER_STREAMCUT - User provides the StreamCut for truncation using Consumption Based Retention.
```

Comment:
> User provides the StreamCut for truncation using Consumption Based Retention.

This should link to the API through which it is provided. 
The streamcut is not "for truncation".

Reply:
That API is not added as part of this PR but as part of another PR. So this javadoc can be updated there.

Reply:
We can't have these in different versions of pervega. It leaves the API in an inconsistent state.
Please combine these PRs.

Reply:
Perhaps these PRs should be going into a feature branch. It would allow us to avoid this sort of inconsistency in master.

Reply:
Just to be able to reference it in the javadoc, I have added this API to this PR now and added a link in the javadoc.